cmd/agent/main.go

```diff
@@ -158,8 +159,28 @@ func main() {
 	}
 	// !Kubernetes
 
-	// Security
+	// Nomad
+	if containerPlatform == agent.PlatformNomad {
+		advertiseAddr, err = net.GetLocalIP()
+		if err != nil {
+			log.Fatalf("[ERROR] [main,nomad] [message: Unable to retrieve local IP associated to the agent] [error: %s]", err)
+		}
+
+		nomadConfig.NomadAddr = goos.Getenv(agent.NomadAddrEnvVarName)
+		if nomadConfig.NomadAddr == "" {
+			log.Fatalf("[ERROR] [main,nomad] [message: Unable to retrieve environment variable NOMAD_ADDR]")
+		}
 
+		nomadConfig.NomadToken = goos.Getenv(agent.NomadTokenEnvVarName)
+		if nomadConfig.NomadToken == "" {
```

Comment:
@mcpacino 

I think nomad token should be optional, no?

also, to get env vars, you might consider use kingpin (and also receive it as a flag). see https://github.com/portainer/agent/blob/ec007609438be768aa2cb9e5a6668a32874b6459/os/options.go#L38

Reply:
yes, there is a ticket to make it optional